source/configuration/sharding.txt

```diff
@@ -0,0 +1,161 @@
+
+.. meta::
+   :keywords: ruby framework, odm, code example
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+Sharding is a way to distribute your data across multiple machines. MongoDB
+uses sharding to support deployments with large data sets and high
+throughput operations. In this guide, you can learn how to configure sharding in
+your {+odm+} application.
+
+Declare Shard keys
+------------------
+
+MongoDB uses shard keys to distribute a collection's documents across
+shards. A shard key consists of one or more fields in a document. In your
+{+odm+} application, you can declare a shard key when you create a model by
+using the ``shard_key`` macro.
+
+The following example creates a ``Person`` class with a shard key on the
+``ssn`` field:
+
+.. literalinclude:: /includes/configuration/sharding.rb
+   :language: ruby
+   :start-after: # start-shard-key
+   :end-before: # end-shard-key
+   :emphasize-lines: 6
+
+.. note::
+
+   To shard a collection, the collection must have an index that starts with the
+   shard key. You can use {+odm+}'s index management functionality to create the
+   index. To learn more about index management with {+odm+}, see the :ref:`Index
+   Management <mongoid-indexes>` guide.
+
+If a model declares a shard key, {+odm+} expects the respective collection to be
+sharded with the specified shard key. When {+odm+} reloads models, it provides
+the shard key with the ``_id`` field to the ``find`` command to improve query
+performance. If the collection is not sharded with the specified shard key,
+queries might not return the expected results.
+
+You can declare shard keys by using either the standard MongoDB
+syntax, or by using a shorthand syntax.
+
+Standard Syntax
+~~~~~~~~~~~~~~~
+
+The standard syntax follows the format
+of the ``mongosh`` :manual:`shardCollection()
+</reference/method/sh.shardCollection>` method, and allows you to specify the
+following types of shard keys:
+
+- Ranged keys
+- Hashed keys
+- Compound keys
+
+The standard syntax also allows you to specify collection and sharding options.
+
+The following example creates a shard key on the ``ssn`` field in each of the
+preceding formats:
+
+.. literalinclude:: /includes/configuration/sharding.rb
+   :language: ruby
+   :start-after: # start-shard-key-formats
+   :end-before: # end-shard-key-formats
+
+Shorthand Syntax
+~~~~~~~~~~~~~~~~
+
+The shorthand syntax allows you to declare a shard key by specifying only the
+field name. This syntax supports only ranged and compound shard keys, and does not allow you
+to specify collection or sharding options.
+
+The following example creates a ranged and a compound shard key:
+
+.. literalinclude:: /includes/configuration/sharding.rb
+   :language: ruby
+   :start-after: # start-shard-key-shorthand
+   :end-before: # end-shard-key-shorthand
+
+Specify Associated and Embedded Fields
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
S: this is placed as a sibling heading of standard/shorthand syntax, which sort of suggests it's a third alternative to those. i'm guessing that's not the case? if not, you could take standard/shorthand out of their own subheadings and just put them under a single heading called 'Syntax' or something.